### PR TITLE
Refactoring related modules for readability and usage

### PR DESCRIPTION
**Pull request by richaagarwal**
This PR attempts to clean up some of the code around related modules by
- Moving functions onto the CFGOVPage class when possible, instead of as helper functions
- Rewriting `related_posts` so that it's more readable. Does not change functionality 

All tests still pass

### CODE REVIEW

File: cfgov/v1/models/base.py [hunk 1]
```diff
@@ -149,73 +149,74 @@ def generate_view_more_url(self, request):
...
-                    AbstractFilterPage.objects.live().filter(
-                        search_query
-                    ).distinct().exclude(id=self.id).order_by(
-                        '-date_published'
-                    )
-                )
+        specific_categories = block.value['specific_categories']
+
+        for slug, should_get_related_posts in search_types:
+            if should_get_related_posts:
```
Comment (anselmbradford) [inline cfgov/v1/models/base.py]: Could this be a guard clause? So return immediately if `should_get_related_posts` does not exist. I don't know what that would look like in python.

File: cfgov/v1/models/base.py [hunk 2]
```diff
@@ -149,73 +149,74 @@ def generate_view_more_url(self, request):
...
+        for tag in self.specific.tags.all():
+            tag_link = {'text': tag.name, 'url': ''}
+            if id is not None and filter_page is not None:
+                param = '?filter' + str(id) + '_topics=' + tag.slug
+                tag_link['url'] = relative_url + param
+            tags['links'].append(tag_link)
+        return tags
+
+    def get_filter_data(self):
+        for ancestor in self.get_ancestors().reverse().specific():
```
Comment (anselmbradford) [inline cfgov/v1/models/base.py]: Should `self.get_ancestors().reverse().specific()` go in its own variable before the loop? It looks like each method will get called each loop iteration.

**Comment by anselmbradford** [comment]
Passes tox tests. Probably would help to get a bewd review on the actual code. 👍

File: cfgov/v1/models/base.py (hunk 2 above)
Comment (richaagarwal) [reply cfgov/v1/models/base.py]: It's called in the outermost loop so it's only executed once

File: cfgov/v1/models/base.py (hunk 1 above)
Comment (richaagarwal) [reply cfgov/v1/models/base.py]: I changed this in another way, but one option could've been to do `if not should_get_related_posts: continue` and then have all the code below that indented to the left.

**Pull request merged**